lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -334,7 +334,11 @@ defmodule Mix.Compilers.Elixir do
 
   defp update_stale_sources(sources, changed) do
     # Store empty sources for the changed ones as the compiler appends data
-    Enum.reduce(changed, sources, &List.keystore(&2, &1, source(:source), source(source: &1)))
+    Enum.reduce(
+      changed,
+      sources,
+      &List.keystore(&2, &1, source(:source), source(source: &1, size: :filelib.file_size(&1)))
```

Comment:
We already have this information from a couple lines above. I would prefer to re-use it instead of reaching the file system again.

Reply:
👍